pandas/core/arrays/categorical.py

```diff
@@ -2349,9 +2349,9 @@ def _can_hold_na(self):
 
     @classmethod
     def _concat_same_type(self, to_concat):
-        from pandas.core.dtypes.concat import concat_categorical
+        from pandas.core.dtypes.concat import union_categoricals
 
-        return concat_categorical(to_concat)
+        return union_categoricals(to_concat)
```

Comment:
IIRC only a relatively small part of the logic of concat_categorical/union_categoricals is actually needed here.  I'd prefer for that to live here and for union_categoricals to call it, rather than the other way around (since union_categoricals handles a lot of cases).  Could be considered orthogonally to this PR.

Reply:
>  I'd prefer for that to live here and for union_categoricals to call it, rather than the other way around

Yes, that's indeed a good idea (union_categoricals does way much more that is all not needed here)

Reply:
are you planning to update this, or is that a topic for a separate PR?